File: 2265-partition-array-according-to-given-pivot/partition-array-according-to-given-pivot.cpp
class Solution {
public:
    vector<int> pivotArray(vector<int>& nums, int pivot) {
        vector<int>ans;
        int cnt=0;
        int n = nums.size();
        for(int i=0;i<n;i++){
            if(nums[i]==pivot)
                cnt++;
            if(nums[i]<pivot)
                ans.push_back(nums[i]);
        }

        for(int i=0;i<cnt;i++){
            ans.push_back(pivot);
        }

        for(int i=0;i<n;i++){
            if(nums[i]>pivot)
                ans.push_back(nums[i]);
        }

        return ans;
    }
};

File: 2265-partition-array-according-to-given-pivot/partition-array-according-to-given-pivot.java
class Solution {
    public int[] pivotArray(int[] nums, int pivot) {
        int cnt = 0;
        int n = nums.length;
        int [] ans = new int[n];

        int j = 0;
        for(int i = 0; i < n; i++){
            if(nums[i] == pivot)
                cnt++;
            if(nums[i] < pivot){
                ans[j] = nums[i];
                j++;
            }
        }

        for(int i = 0; i < cnt; i++){
            ans[j] = pivot;
            j++;
        }

        for(int i = 0; i < n; i++){
            if(nums[i] > pivot){
                ans[j] = nums[i];
                j++;
            }
        }

        return ans;
    }
}

